File: day-03/lua/part1.lua
#!/usr/bin/env lua5.4

local function parse(inp)
    local num = {}
    for str in io.lines(inp) do
        num[#num+1] = str
    end
    return num
end

local data = parse(arg[1])
local slope = {3, 1}

local numRows = #data
local numCols = string.len(data[1])

local trees = 0
local x = 0

for y=1,numRows,slope[2] do
    if data[y]:sub(x+1, x+1) == '#' then
        trees = trees + 1
    end
    x = (x + slope[1]) % numCols
end

print("trees "..trees)

File: day-03/python/part1.py
#!/usr/bin/env python3

import sys

data = [x.strip() for x in open(sys.argv[1], 'r').readlines()]

slope = (3, 1)  # right, down

numRows = len(data)  # changes between the example and actual input
numCols = len(data[0])  # ^

# m = y2 - y1 / x2 - x1 -> basically y/x

trees = 0
x = 0

for y in range(0, numRows, slope[1]):  # -> step right by factor of right (in this case, step right by a factor)
    if data[y][x] == '#':  # this is not a 2d array, it is checking the y and x-axis for a "tree"
        trees += 1
    x = (x + slope[0]) % numCols  # allow for "looping" around on the x-axis
    
print(f"trees: {trees}")


